app/bundles/EmailBundle/DependencyInjection/EnvProcessor/MailerDsnEnvVarProcessor.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Mautic\EmailBundle\DependencyInjection\EnvProcessor;
+
+use Symfony\Component\DependencyInjection\EnvVarProcessorInterface;
+use Symfony\Component\Mailer\Transport\Dsn;
+
+class MailerDsnEnvVarProcessor implements EnvVarProcessorInterface
```

Comment:
@escopecz @RCheesley If `Dsn` setting is faulty, this processor will make the default `Dsn` to null, otherwise Mautic will break. Null means Mautic will not send emails, this needs to be well documented. I also show a note in red to tell the user there is an issue in the config. **We need a business decision on what should be done**, we will face a similar case with `messenger`

Reply:
I think that is good to show it in the config, but I wonder if something so critical as this should also be shown in a bar message (like we have with the bar focus items) across Mautic until it is resolved? It is not the kind of thing that you want to miss and I would be concerned that it might go unnoticed if it is screwed up for example during a command line installation where the user might not have actually edited/saved the config themselves?

Reply:
I think we should go the other way. Use the DSN field in the config instead of having special field for each information separately. And write a migration to compose the DNS string from the existing email config values. WDYT?

Reply:
@escopecz good suggestion, but what would you do if the user inserts wrong values for some reason? the DSN string must be right all the time otherwise it will break

Reply:
We should validate the DSN string. And the values in it must be right the same way as the values in each separate field must be right.

Reply:
1. DSN validation was added.
2. When an invalid DSN is put via `local.php`, Mautic will no longer crash.